ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -0,0 +1,376 @@
+
+    private var contentViewController: UIViewController
+
+    private var contentView: UIView { contentViewController.view }
+
+    private lazy var panGestureRecognizer: UIPanGestureRecognizer = UIPanGestureRecognizer(target: self, action: #selector(handlePan))
+
+    private var translationAnimator: UIViewPropertyAnimator?
+
+    private var currentOffsetFromBottom: CGFloat {
+        -bottomSheetOffsetConstraint.constant
+    }
+
+    private var collapsedOffsetFromBottom: CGFloat {
+        collapsedContentHeight + (isExpandable ? ResizingHandleView.height : 0.0)
+    }
+
+    private var expandedOffsetFromBottom: CGFloat {
+        return bottomSheetView.frame.height - Constants.Spring.overflowHeight
+    }
+
+    private struct Constants {
+        // Maximum offset beyond the normal bounds with additional resistance
+        static let maxRubberBandOffset: CGFloat = 20.0
+        static let minRubberBandScaleFactor: CGFloat = 0.05
+
+        // Swipes over this velocity ignore proximity to the collapsed / expanded offset and fly towards
+        // the offset that makes sense given the swipe direction
+        static let directionOverrideVelocityThreshold: CGFloat = 150
+
+        // Minimum padding from top when the sheet is fully expanded
+        static let minimumTopExpandedPadding: CGFloat = 25.0
+        static let defaultCollapsedContentHeight: CGFloat = 75
+
+        struct Spring {
+            // Spring used in slow swipes - no oscillation
+            static let defaultDampingRatio: CGFloat = 1.0
+
+            // Spring used in fast swipes - slight oscillation
+            static let oscillatingDampingRatio: CGFloat = 0.8
+
+            // Swipes over this velocity get slight spring oscillation
+            static let flickVelocityThreshold: CGFloat = 800
+
+            static let maxInitialVelocity: CGFloat = 40.0
+            static let animationDuration: TimeInterval = 0.4
+
+            // Off-screen overflow that can be partially revealed during spring oscillation or rubber banding (dragging the sheet beyond limits)
+            static let overflowHeight: CGFloat = 50.0
+        }
+    }
+}
+
+extension BottomSheetController: UIGestureRecognizerDelegate {
+
+    public func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldBeRequiredToFailBy otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        return gestureRecognizer == panGestureRecognizer && otherGestureRecognizer == hostedScrollView?.panGestureRecognizer
+    }
+
+    public func gestureRecognizerShouldBegin(_ gestureRecognizer: UIGestureRecognizer) -> Bool {
+        guard let scrollView = hostedScrollView, let panGesture = gestureRecognizer as? UIPanGestureRecognizer else {
+            return true
+        }
+
+        let scrolledToTop = scrollView.contentOffset.y <= 0
+        let panningDown = panGesture.velocity(in: view).y > 0
+        let fullyExpanded = currentOffsetFromBottom >= expandedOffsetFromBottom
```

Comment:
Should we check this first and make the checks sequential so we don't have to calculate contentOffset or velocity if we don't need to?